spock-specs/src/main/groovy/org/spockframework/specs/extension/SnapshotExtension.groovy

```diff
@@ -17,24 +21,63 @@ class SnapshotExtension implements IAnnotationDrivenExtension<Snapshot> {
   }
 
   @Override
-  void visitFieldAnnotation(Snapshot annotation, FieldInfo field) {
-    Assert.that(field.type.isAssignableFrom(Snapshotter), "Field must be of type Snapshotter")
-    field.parent.bottomSpec.addSetupInterceptor(new SnapshotInterceptor(annotation, field))
+  void visitSpecAnnotation(Snapshot annotation, SpecInfo specInfo) {
+    specInfo.features.each { visitFeatureAnnotation(annotation, it) }
+  }
+
+  @Override
+  void visitFeatureAnnotation(Snapshot annotation, FeatureInfo feature) {
+    def snapshotterFields = feature.spec.allFields.findAll {
+      ((it.type == Snapshotter) || (it.type.isAssignableFrom(Snapshotter) && (it.name == 'snapshotter'))) &&
+        !it.static && !it.shared
+    }
+    def snapshotterParameters = feature.featureMethod.parameters.findAll {
+      ((it.reflection.type == Snapshotter) || (it.reflection.type.isAssignableFrom(Snapshotter) && (it.name == 'snapshotter'))) &&
```

Comment:
I don't think that we need to support `(it.reflection.type.isAssignableFrom(Snapshotter) && (it.name == 'snapshotter')` as it should always be strongly typed in this codebase.